### ISSUE TITLE
Wrong CA1416 warning for If when use == or is

### ISSUE THREAD

**Issue by TheBigNeo** [dotnet/roslyn#58390]
**What is the expected result?**
No `CA1416` warning for `if` when use `== true` or `is true` or `== false` or `is false`

**What steps will reproduce the issue?**
```c#
public static void Func2(Bitmap bitmap)
{
    {
        if (OperatingSystem.IsWindows() is false)
        {
            return;
        }
        bitmap.Save(new MemoryStream(), ImageFormat.Png);
    }
    if (OperatingSystem.IsWindows() == true)
    {
        bitmap.Save(new MemoryStream(), ImageFormat.Png);
    }
    
    if (OperatingSystem.IsWindows() is true)
    {
        bitmap.Save(new MemoryStream(), ImageFormat.Png);
    }
}
```

**What happens instead?**
Rider shows a warning 
![image](https://user-images.githubusercontent.com/24251966/146498739-d5d12f70-450f-4f55-9a3f-a737f81b035c.png)

**Attachments:**
Sample project: [SampleProjectOperatingSystem.IsWindows.zip](https://github.com/dotnet/roslyn/files/7732460/SampleProjectOperatingSystem.IsWindows.zip)

Version: RD-213.5744.299
Timezone: Europe/Berlin

valuation: false
Environment: RD-213.5744.299, JRE 11.0.13 7-b1751.19x64 JetBrains s.r.o., OS Windows 10(amd64) v10.0 , screens 3200.0x1800.0, 4300.0x1800.0, 4300.0x1800.0
.NET Framework 4.0.30319.42000
